### PR TITLE
Add Block & Refusal justifications

### PR DESCRIPTION
**Pull request by Remg**

### CODE REVIEW

File: templates/admin/certification_request/refuse.html.twig [hunk 1]
```diff
@@ -32,3 +82,22 @@
...
+
+{% block javascripts %}
+    {{ parent() }}
+
+    <script type="text/javascript">
+        $(document).ready(function() {
+            var refusalCustomReason = $('#refusal-custom-reason');
+
+            $('#{{ form.reason.vars.id }}').change(function() {
+                if ('other' == $(this).val()) {
```
Comment (ottaviano) [inline templates/admin/certification_request/refuse.html.twig:94]: `constant('App\\Adherent\\CertificationRequestBlockCommand::BLOCK_REASON_OTHER')`

File: templates/admin/certification_request/show.html.twig [hunk 2]
```diff
@@ -184,6 +184,42 @@
...
+                                        ({{ object.blockComment }})
+                                    {% endif %}
+                                </p>
+                            {% endif %}
+
+                            {% if object.refusalReason %}
+                                <p class="form-text text-muted text-center">
+                                    Refusée:
+                                    <b>
+                                        {% if constant('App\\Adherent\\CertificationRequestRefuseCommand::REFUSAL_REASON_OTHER') == object.refusalReason %}
```
Comment (ottaviano) [inline templates/admin/certification_request/show.html.twig:210]: as an alternative for avoiding to do this check many times: `object.refusedWithOtherRaison(): bool` helper

File: src/Adherent/Certification/CertificationRequestBlockCommand.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+     * @Assert\Choice(choices=CertificationRequestBlockCommand::BLOCK_REASONS)
+     */
+    private $reason;
+
+    /**
+     * @var string|null
+     *
+     * @Assert\Length(max=500)
+     * @Assert\Expression(
+     *     expression="constant('App\\Adherent\\CertificationRequestBlockCommand::BLOCK_REASON_OTHER') !== this.getReason() or value",
```
Comment (kevin-verschaeve) [inline src/Adherent/Certification/CertificationRequestBlockCommand.php:34]: are you sure about this condition ?
Shouldn't it be 
```suggestion
     *     expression="constant('App\\Adherent\\CertificationRequestBlockCommand::BLOCK_REASON_OTHER') === this.getReason() and not value",
```
?

File: src/Adherent/Certification/CertificationRequestRefuseCommand.php [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+     * @Assert\Choice(choices=CertificationRequestRefuseCommand::REFUSAL_REASONS)
+     */
+    private $reason;
+
+    /**
+     * @var string|null
+     *
+     * @Assert\Length(max=500)
+     * @Assert\Expression(
+     *     expression="constant('App\\Adherent\\CertificationRequestRefuseCommand::REFUSAL_REASON_OTHER') !== this.getReason() or value",
```
Comment (kevin-verschaeve) [inline src/Adherent/Certification/CertificationRequestRefuseCommand.php:34]: same question

**Pull request merged**